integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/command/SICreationCommand.scala

```diff
@@ -253,6 +253,17 @@ private[sql] case class CarbonCreateSecondaryIndexCommand(
         throw new ErrorMessage(
           s"Table [$tableName] under database [$databaseName] is already an index table")
       }
+      // creation of index on long string columns are not supported
+      for (indexColName <- indexModel.columnNames) {
+        for (dim <- dims) {
+          if (indexColName.equals(dim.getColName) && dim.getDataType.getName.equals("VARCHAR")) {
+            throw new ErrorMessage(
+              s"one or more index columns specified cannot be of Long String property" +
+              s" column in table $databaseName.$tableName")
+          }
+        }
+      }
+
```

Comment:
instead of using a standard for loop, can change to this,

> if (dims.filter(dimension => indexModel.columnNames 
>         .contains(dimension.getColName)) 
>         .map(_.getDataType) 
>         .exists(dataType => dataType.equals(DataTypes.VARCHAR)) ) { 
>         throw new ErrorMessage( 
>           s"one or more index columns specified contains long string column" + 
>           s" in table $databaseName.$tableName. SI cannot be created on long string columns") 
>       }
> 

 in a more functional way

Reply:
done